res/enkf/active_list.py

```diff
@@ -35,6 +37,20 @@ def __init__(self):
     def getMode(self):
         return self._get_mode()
 
+    def getActiveIndexList(self):
+        """
+        Returns a list of indices corresponding to active parameters for a parameter node
+        Requires PARLY_ACTIVE as ActiveMode, else return None
+        """
+        mode = self.getMode()
+        if mode == ActiveMode.PARTLY_ACTIVE:
+            size = self._asize(0)
+            c_ptr = self._get_active_index_list()
+            index_list = [c_ptr[i] for i in range(size)]
+            return index_list
+        else:
+            return None
```

Comment:
I know this is how the C-implementation works, but is it possible to find the size and return a list of 0/1 for INACTIVE/ACTIVE?

Reply:
To find the size is more tricky for ACTIVE. For INACTIVE it should be 0. The index list has the advantage when the full size is large and the number of active is small. Are there any benefits to return a 0/1 array of full size in practice? They way I find the full size: For GEN_PARAM type parameters, I use the gen_data_config =param node.getDataModelConfig() og data_size = gen_data_config.get_initial_size()
For GEN_KW type parameters, I use kw_config_model = node.getKeywordModelConfig() and parameter_list = kw_config_model.getKeyWords().strings and size = len(parameter_list)
So given only the active_list_object, I don't know where it comes from. But in a context where the node is known, it is possible like what I did above.

Reply:
The reason for returning a 0/1 array would be that we dont special case everything, then there is no need to check every time if there exists a list or not. Even for inactive, the list is technically length 0 at the moment, however if you change but onel to be active the behavior changes a lot, the length is no longer the same, for the same ActiveList.

Reply:
So looking a bit more into it, it looks like the underlying `index_list` does not even exists if it is all active? Or it does not have to?

Reply:
When using ActiveList either related to parameters(GEN_KW, GEN_PARAM) or to observations (GEN_OBS), I always go through all items and use addActiveIndex for those that are active and I always test on ActiveMode.PARTLY_ACTIVE so in localisation setting this is not a problem.